src/arch/LowLevelWindow/LowLevelWindow_SDL.cpp

```diff
@@ -0,0 +1,258 @@
+        m_pGlContext = SDL_GL_CreateContext(m_pWindow);
+
+        if (!m_pGlContext)
+        {
+            LOG->Info("[LLW_SDL::TryVideoMode] Failed to create GL Context: %s", SDL_GetError());
+            return "Failed to create GL Context - Check logs";
+        }
+
+        SDL_GL_MakeCurrent(m_pWindow, m_pGlContext);
+        SDL_ShowWindow(m_pWindow);
+
+        // New window created
+        bNewDeviceOut = true;
+    } else
+    {
+        bNewDeviceOut = false;
+    }
+
+    // Adjust for any hidpi scaling & prevent division by zero
+    // Note: This is rounded to the nearest 4 decimals to prevent improperly set scale values
+    auto windowScale = std::floor(SDL_GetWindowDisplayScale(m_pWindow) * 1000) / 1000;
+    if (windowScale <= 0.01f)
+    {
+        windowScale = 1.0f;
+    }
+    SDL_SetWindowSize(m_pWindow, static_cast<int>(static_cast<float>(p.width) / windowScale), static_cast<int>(static_cast<float>(p.height) / windowScale));
+
+    // Retrieve the real pixel size in case we have a high-dpi display
+    int widthInPixels = p.width;
+    int heightInPixels = p.height;
+
+    // We want the currentDisplay (monitor), the current display mode (resolution),
+    // and the real window size in pixels (ignoring hidpi)
+    auto currentDisplay = SDL_GetDisplayForWindow(m_pWindow);
+    auto currentDisplayMode = SDL_GetCurrentDisplayMode(currentDisplay);
+    SDL_GetWindowSizeInPixels(m_pWindow, &widthInPixels, &heightInPixels);
+
+    LOG->Info("[LLW_SDL::TryVideoMode] Adjusting for HiDPI scale of %f. Requested: [%dx%d] Real: [%dx%d]", windowScale,
+              p.width, p.height, widthInPixels, heightInPixels);
+
+    // Adjust VSync, fullscreen, and borderless modes if needed
+    SDL_GL_SetSwapInterval(p.vsync);
+    SDL_SetWindowFullscreen(m_pWindow, !p.windowed);
+    SDL_SyncWindow(m_pWindow);
+    SDL_SetWindowBordered(m_pWindow, !p.windowed || !p.bWindowIsFullscreenBorderless);
+
+    // If we're just resizing we should re-centre the window
+    if (!bNewDeviceOut)
+    {
+        SDL_SetWindowPosition(m_pWindow, SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED);
+    }
+
+    if (p.sIconFile)
+    {
+        SetWindowIcon(p.sIconFile);
+    }
+
+    // Set the "Real Width/Height" to the HiDpi values so the scaling is correct
+    m_CurrentVideoParams = ActualVideoModeParams(p, widthInPixels, heightInPixels, false);
+    // Set the refresh rate to the current display's refresh rate
+    // Note: The casting to int can cause refresh rates like 59.<whatever> to be truncated to 59.0.
+    // I'm unsure if we want to just ceil the values here.
+    m_CurrentVideoParams.rate = static_cast<int>(currentDisplayMode->refresh_rate);
+
+    // Success!
+    return "";
+}
+
+/**
+ * Enumerate a list of displays and display resolutions (or modes).
+ * Each display is named, and uses the id: SDL_<order number>.
+ * @param out
+ */
+void LowLevelWindow_SDL::GetDisplaySpecs(DisplaySpecs& out) const
```

Comment:
I'm not sure if this comment belongs here specifically, but in the past all  modes were listed so that you could see that they are available:

![image](https://github.com/user-attachments/assets/0f3faeff-9b6c-4a2b-a4ae-a5ec02490f8e)

now there's only the current one shown, but you can still switch it with left-right. I find it to be confusing:

![image](https://github.com/user-attachments/assets/db1d5b8b-68d6-46c8-931f-98ed17459e4e)